Add to `load_truth()`: ECDC hospitalisations

kathsherratt opened the pull request:
We have added [hospitalisations data](https://github.com/epiforecasts/covid19-forecast-hub-europe/blob/main/data-truth/ECDC/truth_ecdc-Incident%20Hospitalizations.csv) to the European hub and it would be helpful to load this using `covidHubUtils`.

These data are daily from the ECDC and stored in the Euro hub repo. 

With this PR:
- `load_truth()` supports getting ECDC hospitalisation data from a local hub repo 
- ECDC hospitalisation data can be at daily or weekly aggregations
- Some of the existing code for ECDC cases and deaths is corrected (the `week_start` is not the same as the `target_end_date`)

There are a few layers of `if...` statements around getting US hospitalisations which I've lightly edited to make this work. I've checked and pretty sure this doesn't break any existing functionality, but still might be good for someone else to also check against how the function is used in the US hub. Also tagging @nikosbosse for review!

Example use:
```
ecdc_hosp <- load_truth(target_variable = "inc hosp", 
                        truth_source = "ECDC", 
                        data_location = "local_hub_repo", 
                        local_repo_path = here(), 
                        hub = "ECDC")

ecdc_hosp_wk <- load_truth(target_variable = "inc hosp", 
                        truth_source = "ECDC", 
                        temporal_resolution = "weekly",
                        data_location = "local_hub_repo", 
                        local_repo_path = here(), 
                        hub = "ECDC")
```

Happy to edit or accept review suggestions!

nikosbosse reviewed (commented):
Looks good to me!

In general I think having unit tests would be good, but that is more a general remark

Serena-Wang reviewed (commented):
Hi @kathsherratt. Thank you for your help!  

I have some questions about the default behavior for ECDC hub:
1. With this new hosp truth, would the default selection for ECDC hub still be `inc case` and `inc death` at line 117?
2. Would you like to include both `week_start` column and `target_end_date` column in the result? In other functions in this package, we are generally using `target_end_date` (Saturdays). Are there other cases that we should use `week_start` for ECDC hub?

Serena-Wang commented on R/load_truth.R:
```diff
@@ -221,49 +224,55 @@ load_truth <- function (truth_source = NULL,
+        if ("week_start" %in% names(truth)) {
+          truth <- truth %>%
+            dplyr::mutate(week_start = as.Date(week_start),
+                          target_end_date = week_start + 5) %>% # to match epiweek
```
With this change, I think it might break the unit test [here](https://github.com/reichlab/covidHubUtils/blob/2604288565dc0a49e2e4d3feb1446b8950a70af0/tests/testthat/test-load_truth.R#L183). This is checking if all `target_end_dates` from `"ECDC"` `truth_source` are Mondays. We should change it to Saturdays.

kathsherratt commented:
Thanks for having a look Serena!

1 - Good idea, added.

2 - The ECDC week and date formats are a bit of a nightmare! Here's my thinking...
- ECDC data for deaths and cases cover from the "start of the week" and then the total count for that week. 
- This means `target_end_date` would actually be up to `week_start` + 6, Sunday. 
- The previous code re-labelled the `week_start` as `target_end_date` to match the formatting convention of truth data. Unfortunately I think this was probably incorrect as this would have implied the wrong period (the `target_end_date` should be `week_start + 6`)
- I think it's all okay as long as it's clear for which week the ECDC data applies to. I thought this was probably best done by keeping the `week_start` variable.
- I've suggested setting the `target_end_date` as `week_start` + 5, will match with the rest of the hub's "truth week" (just including 1 day extra data).
- I've updated this in the test as well: `week_start` as Monday, `target_end_date` as Saturday.
- This only applies to case and death counts....... the hospital data are daily, so we would want to aggregate them to the same US/epi weeks.

Bisaloo commented on R/load_truth.R:
```diff
@@ -146,6 +143,14 @@ load_truth <- function(truth_source = NULL,
+    # extra checks for truth source if target is inc hosp
+    if ("inc hosp" %in% target_variable) {
+      if (!"ECDC" %in% truth_source) {
+        warning("Warning in load_truth: Incident hopsitalization truth data is only available from ECDC.
```
```suggestion
        warning("Warning in load_truth: Incident hospitalization truth data is only available from ECDC.
```

Serena-Wang commented:
Thanks everyone! I'm merging this pull request now since it passes r cmd check and the majority of the unit tests. There's a minor problem in test-load_truth() because incident hospitalization is added to the default targets for ECDC hub and a particular test case is not updated. I could change it on my end.

sbfnk commented:
Thanks @Serena-Wang!
